examples/jsm/nodes/shadernode/ShaderNodeBaseElements.js

```diff
@@ -108,7 +108,7 @@ export const uniform = ( nodeOrType ) => {
 
 };
 
-export const attribute = ( name, nodeOrType ) => nodeObject( new AttributeNode( name, getConstNodeType( nodeOrType ) ) );
+export const attribute = ( name, nodeType ) => nodeObject( new AttributeNode( name, nodeType ) );
```

Comment:
I think it is better to just make `getConstNodeType` return undefined if input is undefined.

Reply:
What kind of use did you imagine when you implemented this function? For this case, I see only 3 type of usage:
```js
// name only
attribute( 'color' );

// typed
attribute( 'color', 'vec3' );

// typed from primiteve class
attribute( 'color', vec3 );
```

Reply:
It can handle `attribute( 'color' )` (if make that change), `attribute( 'color', 'vec3' )` and `attribute( 'color', vec3() )`. Did not thought about `attribute( 'color', vec3 )`, but it can also be easily added.

Reply:
> Did not thought about attribute( 'color', vec3 ), but it can also be easily added.

Nice, is better use `attribute( 'color', vec3 )` instead of `attribute( 'color', vec3() )` because it's not allocate unnecessary object in memory. `attribute( 'color', vec3() )` shouldn't have support.